From patchwork Sun Jan  7 18:04:02 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [LEDE-DEV,fstools] overlay: fix compilation with glibc
X-Patchwork-Submitter: Roman Yeryomin <roman@advem.lv>
X-Patchwork-Id: 856577
Message-Id: <20180107180402.14368-1-roman@advem.lv>
To: LEDE Development List <lede-dev@lists.infradead.org>
Date: Sun,  7 Jan 2018 20:04:02 +0200
From: Roman Yeryomin <roman@advem.lv>
List-Id: <lede-dev.lists.infradead.org>

glibc fails with:
warn_unused_result [-Wunused-result]
   system("cp -a /tmp/root/upper/* / 2>/dev/null");

Regression intoroduced with 11efbf3b90f031ce634fc38cad07a4aef2985777
Reported here:
https://www.mail-archive.com/lede-dev@lists.infradead.org/msg10502.html

Signed-off-by: Roman Yeryomin <roman@advem.lv>
Acked-by: Florian Fainelli <f.fainelli@gmail.com>
---
 libfstools/overlay.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libfstools/overlay.c b/libfstools/overlay.c
index 0187a55..1d69f5a 100644
--- a/libfstools/overlay.c
+++ b/libfstools/overlay.c
@@ -257,7 +257,8 @@ jffs2_switch(struct volume *v)
 
 		/* try hard to be in sync */
 		ULOG_INFO("syncronizing overlay\n");
-		system("cp -a /tmp/root/upper/* / 2>/dev/null");
+		if (system("cp -a /tmp/root/upper/* / 2>/dev/null"))
+			ULOG_ERR("failed to sync jffs2 overlay\n");
 		break;
 
 	case FS_EXT4:
